nixos/tests/nextcloud/with-objectstore.nix

```
@@ -64,31 +65,39 @@ in {
       export RCLONE_CONFIG_NEXTCLOUD_VENDOR="nextcloud"
       export RCLONE_CONFIG_NEXTCLOUD_USER="${adminuser}"
       export RCLONE_CONFIG_NEXTCLOUD_PASS="$(${pkgs.rclone}/bin/rclone obscure ${adminpass})"
+      "''${@}"
     '';
     copySharedFile = pkgs.writeScript "copy-shared-file" ''
       #!${pkgs.runtimeShell}
+      set -euxo pipefail
       echo 'hello world' | ${pkgs.rclone}/bin/rclone rcat nextcloud:test-shared-file
     '';
 
     diffSharedFile = pkgs.writeScript "diff-shared-file" ''
       #!${pkgs.runtimeShell}
-      export AWS_ACCESS_KEY_ID=${accessKey}
-      export AWS_SECRET_ACCESS_KEY=${secretKey}
-      ${pkgs.awscli2}/bin/aws s3 cp s3://nextcloud /tmp/. --recursive --endpoint-url http://nextcloud:9000 --region us-east-1
-      grep -r "hello world" /tmp
+      set -euxo pipefail
+      diff <(echo 'hello world') <(${pkgs.rclone}/bin/rclone cat nextcloud:test-shared-file)
     '';
   in ''
     start_all()
     nextcloud.wait_for_unit("multi-user.target")
     nextcloud.wait_for_unit("minio.service")
```

```suggestion
    nextcloud.wait_for_unit("nextcloud.service")
```
We can probably drop the check for nextcloud/login below now

---

There's no `nextcloud.service`. Also, fpm being does not imply a successful response (e.g. DB connectivity issues, configuration issues etc.).